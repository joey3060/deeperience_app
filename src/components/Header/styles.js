const React = require('react-native')
const { StyleSheet } = React
module.exports = StyleSheet.create({
  container: {
    flexDirection: 'column',
    flex: 1,
    marginTop: 10,
  },
  header: {
    marginTop: 20,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'transparent',
  },
  mark: {
    height: 100,
    width: 100,
  },
})
