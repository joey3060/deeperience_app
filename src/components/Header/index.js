'use strict'

import React, { PropTypes } from 'react'
import
{
  Text,
  View,
  TouchableHighlight,
} from 'react-native'
import styles from './styles'
import Icon from 'react-native-vector-icons/FontAwesome'

class Header extends React.Component {

  static propTypes = {
    headerText: PropTypes.string,
    back: PropTypes.bool,
  }
  static defaultProps = {
    back: true,
  }
  render() {
    return (
      <View style={styles.container}>
        {
          (() => {
            if (this.props.back) {
              return (
                <TouchableHighlight
                  style={styles.backIcon}
                  onPress={() => this.props.onReturn()}
                >
                  <Icon
                    name={"chevron-left"}
                    size={30}
                    color={"white"}
                  />
                </TouchableHighlight>
              )
            }
          })()
        }
        <Text style={styles.textStyle}>{this.props.headerText}</Text>
      </View>
    )
  }
}

export default Header
