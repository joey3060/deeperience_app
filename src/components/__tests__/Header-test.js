import 'react-native'
import React from 'react'
import Header from '../Header'
import renderer from 'react/lib/ReactTestRenderer'
describe('<Header />', () => {
  it('render if Head_text is empty', () => {
    const Header_Text = ''
    const tree = renderer.create(<Header Header_Text={Header_Text}/>)
    expect(tree).toMatchSnapshot()
  })
  it('render if Head_text not empty', () => {
    const Header_Text = 'test'
    const tree = renderer.create(<Header Header_Text={Header_Text}/>)
    expect(tree).toMatchSnapshot( )
  })
})