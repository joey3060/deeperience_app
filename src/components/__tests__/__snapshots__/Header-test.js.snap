exports[`<Header /> render if Head_text is empty 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F78B6D",
      "justifyContent": "center",
      "marginTop": 20
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    lineBreakMode="tail"
    style={
      Object {
        "color": "#ffffff",
        "fontSize": 20,
        "fontWeight": "bold",
        "lineHeight": 30,
        "marginBottom": 10
      }
    }>
    
  </Text>
</View>
`;

exports[`<Header /> render if Head_text not empty 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F78B6D",
      "justifyContent": "center",
      "marginTop": 20
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    lineBreakMode="tail"
    style={
      Object {
        "color": "#ffffff",
        "fontSize": 20,
        "fontWeight": "bold",
        "lineHeight": 30,
        "marginBottom": 10
      }
    }>
    test
  </Text>
</View>
`;
